client/.generate_component/index.js

```diff
@@ -0,0 +1,54 @@
+const fs = require('fs');
+const { component, sass, sassType, test, barrel } = require('./component_templates.js');
+
+// grab component name from terminal argument
+const [name] = process.argv.slice(2);
+if (!name) throw new Error('You must include a component name.');
+
+const dir = `./src/components/${name}/`;
+
+// throw an error if the file already exists
+if (fs.existsSync(dir)) throw new Error('A component with that name already exists.'); 
+
+// create the folder
+fs.mkdirSync(dir);
+
+function writeFileErrorHandler(err) {
+  if (err) throw err;
+}
+
+// component.tsx
+fs.writeFile(`${dir}/${name}.tsx`, component(name), writeFileErrorHandler);
+// component.module.scss
+fs.writeFile(`${dir}/${name}.module.scss`, sass(name), writeFileErrorHandler);
+// component.module.scss.d.ts
+fs.writeFile(`${dir}/${name}.module.scss.d.ts`, sassType(name), writeFileErrorHandler);
+// test.tsx
+fs.writeFile(`${dir}/${name}.test.tsx`, test(name), writeFileErrorHandler);
+// index.tsx
+fs.writeFile(`${dir}/index.ts`, barrel(name), writeFileErrorHandler);
+
+////////////////
+/// Optional ///
+////////////////
+
+// insert new component into 'components/index.ts file
```

Comment:
Should this whole section be commented out?

Maybe 
```
const OPTIONAL = false;
if (OPTIONAL) {
```
block around whole section?